tools/src/main/scala/scala/scalanative/build/Build.scala

```diff
@@ -53,60 +58,87 @@ object Build {
    *  @return
    *    [[Config#artifactPath]], the path to the resulting native binary.
    */
-  def build(config: Config)(implicit scope: Scope): Path =
+  def build(config: Config)(implicit scope: Scope): Path = {
+    def makeKey(config: Config): Int = {
```

Comment:
This function seems to be big enough to make a private a in Build instead of nesting it here. Also, we could rename it a bit - We do calculate hashCode and assign it to the `inputHash` field,  so `configHashCode`, `getHashCode`, `calcHashCode`, `checkSum` or other similiars might be more appropriate.

Reply:
Renamed as `checkSum` and made it as a private method 👍 https://github.com/scala-native/scala-native/pull/3287/commits/0ed856ef6511b8e4b762f1aeb4fcdf212cb7e359